src/utils/ddb/ddb_vos.c

```diff
@@ -2040,3 +2048,330 @@ dv_dev_replace(const char *db_path, uuid_t old_devid, uuid_t new_devid)
 	vos_self_fini();
 	return rc;
 }
+
+#define GiB                  (1024 * 1024 * 1024)
+#define DEFAULT_SYS_MEM_RSVD (26UL * GiB)
+#define DEFAULT_ENG_MEM_RSVD (1UL * GiB)
+#define DDB_SETUP_BUF_MAX    256
+
+static int
+get_hugepage_total_size(unsigned long *hugepage_total_size)
+{
+	char          line[256];
+	unsigned long hugepage_size = 0;
+	unsigned long hugepage_nr   = 0;
+	int           parse_cnt     = 0;
+	D_ASSERT(hugepage_total_size != NULL);
+
+	FILE *fp = fopen("/proc/meminfo", "r");
+	if (!fp) {
+		D_ERROR("Failed to open /proc/meminfo. %s", strerror(errno));
+		return daos_errno2der(errno);
+	}
+
+	while (fgets(line, sizeof(line), fp)) {
+		if (sscanf(line, "Hugepagesize: %lu kB", &hugepage_size) == 1) {
+			parse_cnt++;
+			continue;
+		}
+		if (sscanf(line, "HugePages_Total: %lu", &hugepage_nr) == 1) {
+			parse_cnt++;
+			continue;
+		}
+	}
+
+	fclose(fp);
+	fp = NULL;
+
+	if (parse_cnt != 2) {
+		D_ERROR("Failed to parse hugepage info from /proc/meminfo. (%lu/%lu)", hugepage_nr,
+			hugepage_size);
+		return -DER_INVAL;
+	} else {
+		D_INFO("Parse hugepage info: (%lu/%lu)", hugepage_nr, hugepage_size);
+	}
+
+	*hugepage_total_size = hugepage_nr * hugepage_size * 1024;
+	return 0;
+}
+
+static int
+auto_calculate_scm_mount_size(unsigned int *scm_mount_size)
```

Comment:
Do we really need to recreate the full control plane formula in the first place? Do we need this automatic mode? It seems overcomplicated and what is even worse it duplicates already existing implementation.

IMHO DDB operator can provide this parameter manually. Or DDB can have some sensible constant default.

Reply:
@janekmi how would the DDB operator know the size ahead of time?